### PR TITLE
fix(upload): support url for file upload (#23)

### PR DESCRIPTION
**Pull request by superrxan**

### CODE REVIEW

**Comment by ngjaying** [comment]
Try to pass codecov target

File: internal/server/rest.go [hunk 1]
```diff
@@ -170,8 +170,50 @@ func createRestServer(ip string, port int, needToken bool) *http.Server {
...
+	if f.Name == "" {
+		return fmt.Errorf("invalid body: name is required")
+	}
+	return nil
+}
+
+func upload(file *fileContent) error {
+	err := getFile(file)
+	if err != nil {
+		return err
```
Comment (ngjaying) [inline internal/server/rest.go:191]: Please try to cover these error condtions as well.

**Review by ngjaying — CHANGES_REQUESTED** [review]
LGTM. Hopefully, we can cover the error conditons to make coverage pass 70%

File: internal/server/rest.go (hunk 1 above)
Comment (superrxan) [reply internal/server/rest.go:191]: ack

**Review by ngjaying — APPROVED** [review]

**Pull request merged**